plutus-metatheory/src/Cost/BuiltinModelAssignment.agda

```diff
@@ -0,0 +1,230 @@
+-- Assignment of models to Builtins
+
+module Cost.BuiltinModelAssignment where
+
+open import Data.Fin using (suc;zero)
+open import Builtin using (Builtin;arity)
+open Builtin.Builtin
+open import Cost.Model using (BuiltinModel;CostingModel;costingCPU;costingMem)
+open CostingModel
+
+assignModel : (b : Builtin) → BuiltinModel (arity b)
```

Comment:
do we need to know the definition of this at compile time? could we just take a function of this type as a parameter, and then make it at runtime by reading the file?

Reply:
I don't think there is a way to dynamically load code at runtime. You would need to read a file with a representation of the data, parse it into a data structure, and then use that data structure to define the function. Which is what I'm trying to avoid because not only it would be a very long data structure, but also it would be another place to touch when you add a new builtin function.

Reply:
Right, I'm suggesting exactly that. I'm not sure why that would be difficult? Wont the data structure be just a dependent map? Which isn't that hard to make? Or you can just return a dependent function? Fundamentally I don't quite see the problem, we just have to write a `Map BuiltinName CostModelDefinition -> Maybe ((b : Builtin) -> BuiltinModel (arity b))` function, which seems doable?

Reply:
I wanted to do it statically, so that if there is any problem with the json file you find about it at compile time.
Maybe not that important, but it's also more efficient.

Reply:
The file is now loaded dynamically at runtime.